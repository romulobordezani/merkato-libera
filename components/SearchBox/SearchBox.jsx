import React, { useState } from 'react';
import Router from 'next/router';
import styles from './SearchBox.scss';

const SearchBox = () => {
  const [q, setQ] = useState('');

  const submitQuery = e => {
    e.preventDefault();
    console.log(q, 'clicked');
    // TODO - Put some decent error feed back here
    if (q !== ``) {
      Router.push(`/items?q=${q}&limit=4`);
    }
  };

  return (
    <div className={styles['search-box']} onSubmit={submitQuery}>
      <form autoComplete="on">
        <input
          className={styles['search-box_field']}
          type="text"
          onChange={event => setQ(event.target.value)}
          placeholder="Nunca dejes de buscar"
        />
        <button className={styles['search-box_submit-button']} type="submit" />
      </form>
    </div>
  );
};

export default SearchBox;
