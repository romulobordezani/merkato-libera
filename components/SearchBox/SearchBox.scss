.search-box {
  width: 100%;
  position: relative;
  overflow: hidden;
  border-radius: 4px;

  &__field {
    background: white;
    width: 100%;
    padding: 7px 60px 9px 15px;
    border: 0 rgba(0,0,0,.2);
    outline: 0;
  }

  &__submit-button {
    position: absolute;
    right: 0;
    top: 0;
    background-image: url(/images/search_icon.svg);
    background-color: #EEEEEE;
    background-repeat: no-repeat;
    background-position: center;
    background-size: 25px;
    border: none;
    cursor: pointer;
    height: 35px;
    padding-left: 27px;
    vertical-align: middle;
    font-size: 18px;
    outline: none;
  }
}
