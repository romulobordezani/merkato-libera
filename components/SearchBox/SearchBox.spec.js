import React from 'react';
import renderer from 'react-test-renderer';

import SearchBox from './SearchBox';

describe('SearchBox', () => {
  it('Snapshot renders default', () => {
    expect(renderer.create(<SearchBox />)).toMatchSnapshot();
  });
});
