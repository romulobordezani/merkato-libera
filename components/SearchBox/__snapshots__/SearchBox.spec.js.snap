// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox Snapshot renders default 1`] = `
<div
  className=" undefined"
  onSubmit={[Function]}
>
  <form
    autoComplete="on"
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Nunca dejes de buscar"
      type="text"
    />
    <button
      type="submit"
    />
  </form>
</div>
`;
